Modules/_datetimemodule.c

```diff
@@ -1939,8 +1944,47 @@ wrap_strftime(PyObject *object, PyObject *format, PyObject *timetuple,
             ptoappend = PyBytes_AS_STRING(freplacement);
             ntoappend = PyBytes_GET_SIZE(freplacement);
         }
+#ifdef Py_NORMALIZE_CENTURY
+        else if (ch == 'Y' || ch == 'G') {
+            /* 0-pad year with century as necessary */
+            PyObject *item = PyTuple_GET_ITEM(timetuple, 0);
+            long year_long = PyLong_AsLong(item);
+
+            if (year_long == -1 && PyErr_Occurred()) {
+                goto Done;
+            }
+            if (year_long > 1000) {
```

Comment:
Year 1000 should be ok, no?

```suggestion
            if (year_long >= 1000) {
```

Reply:
The exclusion of year 1000 was meant specifically for `'%G'` where the beginning of a year may fall in an ISO week of the prior year, but ah no, the year 1000 isn't such a year, so yes you're right that we can include 1000 here. Thanks!

Reply:
No, year 1000 is not OK for %G.

Reply:
@serhiy-storchaka If you try `datetime(1000, 1, 1).strftime('%G')` it returns `'1000'`, so I think it's OK not to 0-pad it.

Reply:
Ok, thanks! That was not immediately obvious to me.

Reply:
Wait; looking more closely at it, year 1000 seems to be ok for %G. Am I missing something? What is the faulty case?

```sh
$ python3 -m calendar 1000 1
    January 1000
Mo Tu We Th Fr Sa Su
       1  2  3  4  5
 6  7  8  9 10 11 12
13 14 15 16 17 18 19
20 21 22 23 24 25 26
27 28 29 30 31
```

Reply:
@blhsing, I did not see your https://github.com/python/cpython/pull/120820#discussion_r1658648747 until now; I agree. ISTM year 1000 is fine. You could probably update that in the Py version also.

Reply:
Well, we a lucky. But it was not obvious.

Should we now add a comment explaining that 1000 is safe for %G, or simply exclude it from fast path?

Reply:
@erlend-aasland Ah yes thanks for the reminder to keep the logics of the Python version in sync.

@serhiy-storchaka Good point. Added comment explaining why year 1000 can go on the fast path for %G then. Thanks.